scripts/dev_cli.sh

```diff
@@ -224,7 +237,7 @@ cmd_tests() {
 	       --volume /dev:/dev \
 	       --volume "$CLH_ROOT_DIR:$CTR_CLH_ROOT_DIR" \
 	       "$CTR_IMAGE" \
-	       ./scripts/run_unit_tests.sh "$@" || exit $?
+	       ./scripts/run_unit_tests.sh "$@" || fix_build_dir_perms $? || exit $?
```

Comment:
fix_build_dir_perms will only run in this case if the tests fail. I think you need to do

```
ret=$?
fix_build_dir_perms
exit $ret
```

Reply:
If the test fails then `fix_build_dir_perms` forwards the test failure and we don't exit. So the final `fix_build_dir_perms` is run afaics.

Reply:
Ah, I missed that at the end, gotcha.